src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,52 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_CAPACITY = 10;
+    private int size;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_CAPACITY];
+        values = (V[]) new Object[MAX_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (areKeysEquals(keys[i], (V) key)) {
+                values[i] = value;
+                return;
+            }
+        }
+        keys[size] = key;
+        values[size] = value;
+        size++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
+            if (areKeysEquals(keys[i], (V) key)) {
+                return values[i];
+            }
+        }
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        return size;
+    }
+
+    public boolean areKeysEquals(K firstKey, V secondKey) {
+        if (firstKey == null && secondKey == null) {
+            return true;
+        }
+        if (firstKey == null || secondKey == null) {
+            return false;
+        }
+        return firstKey.equals(secondKey);
```

Comment:
There is common approach for equals check that is based on fact that null==null:
```suggestion
        return (firstKey == secondKey || firstKey != null && firstKey.equals(secondKey));
```